virttest/utils_misc.py

```diff
@@ -2862,6 +2862,10 @@ def format_windows_disk(session, did, mountpoint=None, size=None,
                 logging.info("Clear readonly bit on 'Disk%s'" % did)
                 session.cmd(set_rw_cmd)
 
+            if re.search(r"Volume.*%s" % fstype, details, re.I | re.M):
```

Comment:
This will work.
But there is a problem regarding this change. When we met reformat disk issue, it means we were using a 'dirty' disk (it may or may not contain data).
Rather than ship format the formatted disk, I think a better solution would be reformat it again to make sure it is clean.

Reply:
Hi hachen,
I got your meaning, but In these cases, we met the reformat disk action after the sub_type test, In fact, we need to test the disk is ok(read/write works normal) after the sub_type test finished, so here we need to check the disk has correct partition and fstype is ok, no need to re-format this disk actually. if it is re-formatted to clear then to do iotest, our test will lose it's meaning~

Reply:
@peixiu you should not assume that this regular function could only be called in some certain case. I agree with @hereischen, that means it needs some tweak, how about:
```python
def format_windows_disk(session, did, mountpoint=None, size=None,
                        fstype="ntfs", force=False):
    ...
    if re.search(r"Volume.*%s" % fstype, details, re.I | re.M) and not force:
        logging.info("Disk%s has been formated, cancel format" % did)
        continue

    # Do force format
    ...
```

Reply:
ok, xuhan, thanks for your advice, will update the code~